### PR TITLE
Proposing Falco for graduation

### PR DESCRIPTION
**Pull request by leodido**
This is a formal proposal to consider Falco for CNCF graduation.

In this PR you can find the proposal document that highlights how Falco has met the graduation criteria as defined by CNCF. 

Please let me know if anything is missing or can be clarified further.

### CODE REVIEW

**Comment by kris-nova** [comment]
strong plus 1 from me, i'll reserve my opinions on exactly *why* i think this should happen for another day. regardless of the reasoning - i do passionately believe this should happen.

**Comment by rikatz** [comment]
+1 from me. 

I've returned to using Falco the last few weeks, the project is solid, the documentation is helpful, the developers/maintainers are accessible, passionate and the community team is great making Falco well suited for graduation and for the de-facto tool in Cloud Native landscape for container security IMO!

**Comment by ssurovich** [comment]
I agree with @rikatz - +1 from me.  Falco has proven itself as an incredibly stable tool that we have made a standard on our clusters.  The team is one of the most helpful OSS project teams I have dealt with, you can see the growth and support from the team on how they have addressed issues over the years, updated documentation, and they add additional support with tools like Sidekick and the Sidekick UI.  This project deserves graduation, I think it deserved consideration a long time ago :)

**Comment by shane-lawrence** [comment]
+1. We've been putting Falco to work @Shopify and have been very happy with the project's evolution. In particular, the maintainers have been uncommonly kind and responsive as they've steered the product to its current status. Especially following https://github.com/falcosecurity/falco/pull/1530, I fully support moving to graduation.

**Comment by cpanato** [comment]
+1
This project reflects a lot in how to build a strong community and build a product that grows taking into consideration feedback from several groups of people.

File: proposals/graduation/falco.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- Downloads:
+  - 24M+ Docker hub pulls (+570% since Incubation)
+  - 30M+ driver downloads from our CDN (+680% since Incubation)
+	- Since Falco compiles a driver for the host when it starts (and fallbacks to download a prebuilt one) this is a very conservative proxy of how many times Falco is installed
+- Top 3 media mentioned Incubated CNCF project for last 3 months (January, February, March 2021)
+- Keynote mention at Kubecon+CloudNativeCon 2020 -- Constance Caramolis ([youtube](https://www.youtube.com/watch?v=uNpZA5fz5o8))
+- End user talk at Kubecon 2020 -- Shane Lawrence, Shopify ([youtube](https://www.youtube.com/watch?v=rBqBrYESryY))
+- End user talk at Kubecon 2020 -- Natch Ruengsakulrach & Eric Hollis, MathWorks ([slides](https://static.sched.com/hosted_files/kccncna20/aa/KubeCon_NA_Virtual_2020-Cyber_Kill_Chain_Falco.pdf))
+- Inclusion in CNCF CKS certification
+- Several vendors have built commercial offerings on top of Falco (e.g. [GitLab](https://about.gitlab.com/blog/2020/08/18/how-gitlab-can-help-you-secure-your-cloud-native-applications/), [SumoLogic](https://www.sumologic.com/solutions/kubernetes/))
```
Comment (Rajakavitha1) [inline proposals/graduation/falco.md:25]: - To enable wider adoption of the product the Falco website content is translated in Chinese, Korean, Japanese, and Malayalam languages.

Comment (Rajakavitha1) [reply proposals/graduation/falco.md:25]: A huge +1 from me!!!

File: proposals/graduation/falco.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- The Falco organization is evolving from a single tool into an ecosystem of projects, some of which are gaining substantial traction (e.g. [falcosidekick](https://github.com/falcosecurity/falcosidekick): 390K+ Docker hub pulls, [falco-exporter](https://github.com/falcosecurity/falco-exporter): 260K+ Docker hub pulls)
+
+## Graduation State Criteria
+
+### ✅ Have committers from at least two organizations.
+
+Since incubation, the amount of contributions coming from organizations other than Falco original creator (Sysdig) has grown steadily. During the last year, around 45% of the contributions came from a diverse group of committers that includes Innoteam, Amazon, Samsung, IBM, Mercari, Red Hat and many individual contributors. More details are available [here](https://falco.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%20year&var-metric=contributions).
+
+[Maintainers](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml) of [Falco projects](https://github.com/falcosecurity) also come from a wide set of companies (Amazon, IBM, VMWare, RedHat, Sysdig, Qonto, move:elevator GmbH, Mercari, Timber).
```
Comment (TheFoxAtWork) [inline proposals/graduation/falco.md:34]: SIG-Security would like to raise a concern regarding the vendor independence of the Falco project.  The[ Incubation Due Diligence document](https://docs.google.com/document/d/1TJCzW8dQ6858lw2UNY-H5LMnvEd4GzwjuOcDInimeyA/edit?ts=5dacfd96#heading=h.378jkvcve1nq) shows many items that are indicated to be resolved prior to graduation.  Reviewing the existing codebase highlights several areas without independence but also [reflect efforts of that transition](https://github.com/falcosecurity/driverkit/commit/a4924530ba68673d661db9652595768e66ba9512).  This in turn causes confusion to the community as it is not clear what the scope of Falco is.  Organizations may not simply deploy Falco as is, rather they must also deploy sysdig to reap the value of the project.

If the separation were to be completed, the existing security audit of 2019 may no longer be relevant to the project as significant changes may have occurred.  The scheduled security audit for later this year would be beneficial, provided the separation were completed.

CC @lizrice & @justincormack

**Review by TheFoxAtWork — COMMENTED** [review]
SIG-Security would like to raise a concern regarding vendor independence (from due diligence doc) and its impact of the existing security audit of the project.

File: proposals/graduation/falco.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+### Link to Incubation Due Diligence(DD) Document
+
+Incubation DD document can be found [here](https://docs.google.com/document/d/1TJCzW8dQ6858lw2UNY-H5LMnvEd4GzwjuOcDInimeyA/edit?ts=5dacfd96#heading=h.378jkvcve1nq).
+
+### Address any concerns or recommendations from the SIG and/or TOC sponsor(s) from the DD Document
+
+One important concern emerged during the incubation due diligence process was related to the codebase’s vendor independence: some of the Falco key data collection dependencies (libsinsp, libscap, kernel module and eBPF probe) were coming from an OSS repository under the Sysdig Inc. GitHub organization.
+
+To address this issue, Sysdig has since contributed those components to the CNCF via the Falcosecurity organization (now available at [falcosecurity/libs](https://github.com/falcosecurity/libs)). Read more on the CNCF blog [here](https://www.cncf.io/blog/2021/02/24/sysdig-contributes-falcos-kernel-module-ebpf-probe-and-libraries-to-the-cncf/).
```
Comment (lizrice) [inline proposals/graduation/falco.md:76]: Seems like there are still [a lot of Sysdig references](https://github.com/search?q=org%3Afalcosecurity+sysdig) - I appreciate that some of these are completely legitimate but almost 500 in the code seems like a lot, please could you comment on that?

Comment (caniszczyk) [reply proposals/graduation/falco.md:76]: Just a reminder for the Falco team, CNCF has website guidelines on how you can refer to the progenitor organization: https://github.com/cncf/foundation/blob/master/website-guidelines.md

**Comment by leodido** [comment]
Hi @lizrice,
Thank you for pointing this out - there are a few reasons why some sysdig references are still around:

1. Most of them (260) are related to the [libraries](https://github.com/falcosecurity/libs/search?q=sysdig) that have recently been [contributed to the CNCF](https://www.cncf.io/blog/2021/02/24/sysdig-contributes-falcos-kernel-module-ebpf-probe-and-libraries-to-the-cncf/).

   In order for this donation to happen, a [dedicated action plan](https://github.com/falcosecurity/falco/blob/master/proposals/2021019-libraries-donation.md) was discussed and approved within the Falco community between Nov 2020 and Jan 2021. As per this plan, we decided not to disrupt the original codebase and git history all at once, but instead to distribute minor refactoring efforts over time (see point #3 of the action plan items), and these efforts are still ongoing.

    I would like to point out that the key items related to the ownership of the libraries' code have been prioritized and completed. These include [org governance](https://github.com/falcosecurity/.github/blob/master/GOVERNANCE.md) (under which the contributed libraries fall), [libs licensing](https://github.com/falcosecurity/libs/blob/master/COPYING#L190) and [using those libs to create Falco pre-built drivers](https://github.com/falcosecurity/driverkit/commit/a4924530ba68673d661db9652595768e66ba9512).
    
    We are confident that removing the vast majority of the references to sysdig in the libraries can be accomplished with a relatively small effort, we are committed to doing it, and there is a PR related to it: [libs#35](https://github.com/falcosecurity/libs/pull/35).

2. Some other references (98) are actually part of archived repositories, precisely: 
    - [falco-security-workshop](https://github.com/falcosecurity/falco-security-workshop/search?q=sysdig) (30)
    - [kubernetes-response-engine](https://github.com/falcosecurity/kubernetes-response-engine/search?q=sysdig) (21)
    - [katacoda-scenarios](https://github.com/falcosecurity/katacoda-scenarios/search?q=sysdig) (21)
    - [falco-operator](https://github.com/falcosecurity/falco-operator/search?q=sysdig) (9)
    - [cloud-native-security-hub](https://github.com/falcosecurity/cloud-native-security-hub/search?q=sysdig) (7)
    - [falco-google-kubernetes-marketplace](https://github.com/falcosecurity/falco-google-kubernetes-marketplace/search?q=sysdig) (4)
    - [falco-aws-firelens-integration](https://github.com/falcosecurity/falco-aws-firelens-integration/search?q=sysdig) (2)
    - [cloud-native-security-hub-backend](https://github.com/falcosecurity/cloud-native-security-hub-backend/search?q=sysdig) (1)
    - [cloud-native-security-hub-frontend](https://github.com/falcosecurity/cloud-native-security-hub-frontend/search?q=sysdig) (1)
    - [profiles](https://github.com/falcosecurity/profiles/search?q=sysdig) (1)
    - [falcosecurity.github.io](https://github.com/falcosecurity/falcosecurity.github.io/search?q=sysdig) (1)

   A dedicated cleanup activity has already been scheduled by the community but postponed since not urgent.

3. Of the remaining 123 references:
    - 46 are contained inside our [website](https://github.com/falcosecurity/falco-website/search?q=sysdig) and mostly refer to the above-mentioned libs before they were contributed
    - 31 are part of [community meeting notes](https://github.com/falcosecurity/community/search?q=sysdig)
    - 22 are into the [main repo](https://github.com/falcosecurity/falco/search?q=sysdig) and most of them are due to disappear as soon as some PRs get merged (e.g. [falco#1552](https://github.com/falcosecurity/falco/pull/1552))
    - 7 are into our [evolution repo](https://github.com/falcosecurity/evolution/search?q=sysdig) which is dedicated to hosting drafts and sandbox projects from our community
    - The remaining 17 items are a mix of old links and legacy mentions

Hope this helps to address your concerns - happy to go deeper into details if needed!

File: proposals/graduation/falco.md (hunk 2 above)
Comment (leodido) [reply proposals/graduation/falco.md:34]: Hello Emily!

I'd like to answer your comment by summarizing the discussion (and the plan we came up with) during our call last week.

To recap:

1. The **current Falco prebuilt drivers** - those installed when using the DEB, RPM, and tarball Falco packages - **are already built from the contributed libs repository** thanks to [driverkit](https://github.com/falcosecurity/driverkit/search?q=DownloadBaseURL) and our [Open Falco Infra](https://prow.falco.org) (Prow) (see [build-drivers](https://github.com/falcosecurity/test-infra/tree/master/images/build-drivers), its [ProwJobs definitions](https://github.com/falcosecurity/test-infra/blob/master/config/jobs/build-drivers/build-drivers.yaml), and an [example run](https://prow.falco.org/log?job=build-drivers-debian-periodic&id=1387602592925749248))
2. There are **2 ongoing PRs to complete the refactoring of the CMakeLists.txt files** both [in Falco](https://github.com/falcosecurity/falco/pull/1552) and [in libs](https://github.com/falcosecurity/libs/pull/30) repository
3. Today, I've finally published 5 **GitHub Security Advisories** on `falcosecurity/libs` **containing detailed information about all the finding of the security audit** done in July 2019, and the relative fixes
      - You can find them [here](https://github.com/falcosecurity/libs/security/advisories?state=published)
4. The findings (and the fixes) of the **security audit** regarding **Falco** only have been **already published** as [Github Security Advisories](https://github.com/falcosecurity/falco/security/advisories)
5. The `falcosecurity/libs` code **doesn't present substantial changes** in the API or in the behavior to the code base that underwent a security audit in July 2019
  The only differences are due to:
      - Fixes to vulnerabilities found during the security audit
      - Fixes on the eBPF driver (mostly to make the eBPF verifier happy on newer Linux kernels)
      - Fixes regarding the `bpf_probe_read_str()` usage
      - Fixes to the handling of syscalls getting relative or absolute file paths
      - Fixes to properly use newer libcurl features
      - Memory leaks fixes
      - Support for tracing the `renameat2` syscall
      - Support for tracing the `fchmodat`, `fchmod`, `chmod` syscalls
      - Support to fetch Kubernetes liveness/readiness probe from related containers (when not available in the container info labels)
      - Async DNS resolution using `c-ares`
      - The userspace API (an experimental API - see pdig driver, or `-u` Falco flag)
      - Decoupling and refactoring code (cosmetic changes) to let the libs contribution happen
      - Refactoring of the CMakeLists.txt files to let the libs contribution happen
      
      We can prepare a deck/presentation to present the libs repository to the SIG-Security, if requested

Roadmap: as you may notice, there's a lot already done to complete the transition, and there's active ongoing work (**point 2**) to complete it that will be complete very soon.

Feel free to reach out if you have any questions. :)

**Comment by robbycochran** [comment]
+1
StackRox utilizes [falcosecurity/libs](https://github.com/falcosecurity/libs) and we would like to contribute upstream patches. We look forward to participating in the Falco community and support the CNCF graduation of Falco.

**Comment by mrbobbytables** [comment]
I'm not a contributor to Falco, but was a heavy end user in my previous role. They've done a great job with building a community around the project 👍 

It's been a little over a month since the last update, but it looks like there has been a good amount of progress towards reducing references:

For the various references to sysdig in the codebase, I did a quick look:
- There are [244 remaining references](https://github.com/search?q=org%3Afalcosecurity+sysdig&type=code), 146 if you omit the archived repos.
- Of those, [24 are references](https://github.com/falcosecurity/libs/search?q=sysdig) in the [falcosecurity/libs](https://github.com/falcosecurity/libs) repo.
- [16 in Falco's primary repo](https://github.com/falcosecurity/falco/search?q=sysdig), with a few being changelog items etc.
- Theres still about the same amount in the [website repo](https://github.com/falcosecurity/falco-website/search?q=sysdig), but a good chunk of those are [blog posts or localizations of content](https://grep.app/search?q=sysdig&filter[repo][0]=falcosecurity/falco-website&filter[path][0]=content/).

If there are concerns about still having a high number of references from the archived projects in the main project org, one potential idea might be to mimic the archive org pattern e.g. [kubernetes-retired](https://github.com/kubernetes-retired) and move the repos there which can act as a very clear signal to users about the state of those projects.

**Comment by fntlnz** [comment]
Hi! Falco maintainer here, just noticed this https://github.com/cncf/toc/pull/671

Should we review our application to follow a new process? Would anyone be able to provide any guidance/help in getting the application updated? 

cc/ @lizrice @amye

**Comment by amye** [comment]
#671 has details on exactly this! 
`The process will change from the current process of submitting a PR and asking for a TOC sponsor to move forward, to a more streamlined process at the time that this PR passes a TOC vote. Projects applying for graduation that do not yet have a TOC sponsor will be moved to this process at the time of an approved vote.`

**Comment by fntlnz** [comment]
I saw that note, I was more focused on the:

> Would anyone be able to provide any guidance/help in getting the application updated?


Just want to make sure we do it right and we don't get lost in processes. We didn't have a chance to connect with people for this and are stuck in a more process driven situation rather than a situation where people are > processes.

I feel like just having a conversation and getting guidance and feedback would help Falco and other projects succeding rather than just following a checklist in this stage.

**Comment by leogr** [comment]
Hi,

Falco maintainer here. Is there any update on the new streamlined process (i.e., #667, #671)? As far as I know, it has not yet been confirmed, and we are still stuck in limbo. It would be beneficial for us to have a guide to understand how to proceed (e.g., if we should fix something or just waiting).

**Comment by amye** [comment]
@leogr: The process has not changed, the changes to the graduating process are still in proposal. My comments from before still apply. 

> #671 has details on exactly this!
> `The process will change from the current process of submitting a PR and asking for a TOC sponsor to move forward, to a more streamlined process at the time that this PR passes a TOC vote. Projects applying for graduation that do not yet have a TOC sponsor will be moved to this process at the time of an approved vote.`

**Comment by lizrice** [comment]
A few thoughts from me 

* I'm confused by the [different](https://github.com/falcosecurity/.github/blob/master/OWNERS) [OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS) files and also [maintainers](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml) file. 

* Two of the most active maintainers moved organization within the last few weeks. That could be great from the point of view of vendor independence, but I'd like to see this given some time to "bed in" and ensure that the project is truly stable. 

* Thanks to @mrbobbytables's update it sounds like there is still some way to go with the removal of Sysdig references?

**Comment by leogr** [comment]
> * I'm confused by the [different](https://github.com/falcosecurity/.github/blob/master/OWNERS) [OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS) files and also [maintainers](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml) file.

The [maintainers.yaml](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml) contains the list of the current maintainers and is generated from the OWNERS files of all projects within [falcosecurity](https://github.com/falcosecurity) GitHub organization. Periodically, it is automatically updated by our infra.

> * Thanks to @mrbobbytables's update it sounds like there is still some way to go with the removal of Sysdig references?

There's an ongoing effort to reduce the reference count. Currently, it already went down from 500+ to 170 (including archived contents and some legitimate references). Is there perhaps any particular area to be cleaned up that we should prioritize?

**Comment by mrbobbytables** [comment]
> Thanks to @mrbobbytables's update it sounds like there is still some way to go with the removal of Sysdig references?

Most of those seem like valid references to sysdig - e.g. 100~ instances are from markdown and most of those are changelog items, meeting notes etc. 

The one standout area is the [sysdig/libs](https://github.com/falcosecurity/libs/) repo, where there are still some carryover code references from when it was migrated, with the caveat that there is a [PR in flight](https://github.com/falcosecurity/libs/pull/41) to take care of most of them.

**Comment by amye** [comment]
Administratively: I'm closing this pull request based on the July 20th meeting, where Falco was discussed and the TOC recommended that Falco reapply in 6 months. 
We look forward to a January application!

**Comment by dims** [comment]
Related traffic:
- https://github.com/falcosecurity/falco/issues/2106
- https://github.com/falcosecurity/evolution/issues/156
- https://github.com/falcosecurity/evolution/issues/157

**Comment by leogr** [comment]
Hey @dims 

We are trying to keep everything within the https://github.com/falcosecurity/evolution repository. 

This :point_down:  is the PR to update the governance document.
https://github.com/falcosecurity/evolution/pull/169

I will keep you posted in case there're other relevant references.

**Pull request closed without merging**